### PR TITLE
Expand docs about adding hello-world to a new track

### PR DESCRIPTION
**Pull request by kytrinyx**
Most people implementing a new track are unfamiliar with all the moving pieces of a track and Exercism's tooling.

This adds a bit more hand-holding so that new volunteers don't need to trawl through the reference documents and draw their own conclusions about how to proceed.

### CODE REVIEW

File: building/tracks/new/add-first-exercise.md [hunk 1]
```diff
@@ -25,3 +22,52 @@ The "Hello, World!" exercise has some special rules applied to it:
...
+
+### Generating required files
+
+To implement the "Hello, World!" exercise, you'll need to create several files.
+Which files exactly is described in the [Practice Exercises documentation](/docs/building/tracks/practice-exercises).
+
+Most of the files can be added automatically by running Configlet's `sync` command:
+
+```
+bin/configlet sync --update --docs --metadata --tests
```
Comment (ErikSchierboom) [inline building/tracks/new/add-first-exercise.md]:
```suggestion
bin/configlet sync --update --yes --docs --metadata --exercise hello-world
bin/configlet sync --update --tests include --exercise hello-world
```

Unfortunately, you can't yet do this in a single command (looks like a bug).

File: building/tracks/new/add-first-exercise.md [hunk 2]
```diff
@@ -25,3 +22,52 @@ The "Hello, World!" exercise has some special rules applied to it:
...
+```
+
+In addition to the generated files, you will to create a test suite, a stub solution that serves as the starting point for the student, and a sample solution that CI uses to ensure that the test suite is in a good state.
```
Comment (ErikSchierboom) [inline building/tracks/new/add-first-exercise.md]: I feel like the bigger reason for having a sample solution is to prove that the exercise _can_ be solved. Maybe something like:

```suggestion
In addition to the generated files, you will to create a test suite, a stub solution that serves as the starting point for the student, and a sample solution that passes all the tests to verify it is possible to solve the exercise (CI will verify this).
```

File: building/tracks/new/add-first-exercise.md [hunk 3]
```diff
@@ -25,3 +22,52 @@ The "Hello, World!" exercise has some special rules applied to it:
...
+
+In order to create these files you need to make some decisions about filenames and file paths.
```
Comment (ErikSchierboom) [inline building/tracks/new/add-first-exercise.md]:
```suggestion
In order to create these files, you need to make some decisions about filenames and file paths.
```

File: building/tracks/new/add-first-exercise.md [hunk 4]
```diff
@@ -25,3 +22,52 @@ The "Hello, World!" exercise has some special rules applied to it:
...
+The rule of thumb is to use names that are idiomatic for the language, and where there are no strong preferences prefer shallower directory structures.
+The sample solution will need to be identifiable by the CI script, so it's advisable to choose a generic basename that all exercises can use, e.g. `example`, `sample`, or `reference-solution`.
+
+### Configuring the exercise
+
+One you've decided on the filenames and paths, edit the `exercises/practice/hello-world/.meta/config.json` file to reflect those choices.
+Also add your GitHub username to the "authors" array.
```
Comment (ErikSchierboom) [inline building/tracks/new/add-first-exercise.md]:
```suggestion
Also add your GitHub username to the `"authors"` array.
```

File: building/tracks/new/add-first-exercise.md [hunk 5]
```diff
@@ -25,3 +22,52 @@ The "Hello, World!" exercise has some special rules applied to it:
...
+```
+bin/configlet sync --update --docs --metadata --tests
+```
+
+In addition to the generated files, you will to create a test suite, a stub solution that serves as the starting point for the student, and a sample solution that CI uses to ensure that the test suite is in a good state.
+
+In order to create these files you need to make some decisions about filenames and file paths.
+The rule of thumb is to use names that are idiomatic for the language, and where there are no strong preferences prefer shallower directory structures.
+The sample solution will need to be identifiable by the CI script, so it's advisable to choose a generic basename that all exercises can use, e.g. `example`, `sample`, or `reference-solution`.
+
```
Comment (ErikSchierboom) [inline building/tracks/new/add-first-exercise.md:68]: I wonder if we should also steer people towards populating the root `config.json`'s `files` property, which can be used as a template for the syncing of the filepaths later on. But maybe that should be a separate item and be part of the next couple of exercises (makes more sense there probably). We could then also point people towards using the same `configlet sync` commands, but those would then also populate the files.

**Review by ErikSchierboom — APPROVED** [review]

File: building/tracks/new/add-first-exercise.md (hunk 1 above)
Comment (kytrinyx) [reply building/tracks/new/add-first-exercise.md]: Ah, I was wondering about that, but hadn't gotten around to asking why it was this way.

File: building/tracks/new/add-first-exercise.md (hunk 5 above)
Comment (kytrinyx) [reply building/tracks/new/add-first-exercise.md:68]: I'm really torn. Getting configlet to sync those files is really handy, but I think coming up with the file names in the context of implementing a concrete exercise is a bit better than coming up with the paths up front and getting configlet to populate the files.

> But maybe that should be a separate item and be part of the next couple of exercises (makes more sense there probably).

I think that's my conclusion as well.

**Pull request merged**